python/pyspark/sql/connect/dataframe.py

```diff
@@ -305,8 +308,12 @@ def _print_plan(self) -> str:
             return self._plan.print()
         return ""
 
-    def collect(self) -> None:
-        raise NotImplementedError("Please use toPandas().")
+    def collect(self) -> List[Row]:
+        pdf = self.toPandas()
```

Comment:
Can this be `None`?

Reply:
yes, returns otional

Reply:
It is because the returned data batch might not contain a `schema`, and the python client implementation uses create Pandas from the data batch (e.g. JSON batch).

If we always include a schema in the data batch, we could drop this `optional`. I am not familiar enough on Pandas for how to append dtypes to the DataFrame from PySpark sql StructType though.